### PR TITLE
[FLINK-9928] Add LOG2 function for table/sql API

### PR DESCRIPTION
**Pull request by yanghua**
## What is the purpose of the change

*This pull request adds LOG2 function for table/sql API*

## Brief change log

  - *Add LOG2 function for table/sql API*

## Verifying this change

This change is already covered by existing tests, such as *ScalarFunctionsTest#testLog2*.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (**yes** / no)
  - If yes, how is the feature documented? (not applicable / **docs** / JavaDocs / not documented)

### CODE REVIEW

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/sql/ScalarSqlFunctions.scala [hunk 1]
```diff
@@ -67,6 +67,15 @@ object ScalarSqlFunctions {
       OperandTypes.family(SqlTypeFamily.NUMERIC, SqlTypeFamily.NUMERIC)),
     SqlFunctionCategory.NUMERIC)
 
+  val LOG2 = new SqlFunction(
+    "LOG2",
+    SqlKind.OTHER_FUNCTION,
+    ReturnTypes.DOUBLE_NULLABLE,
+    null,
+    OperandTypes.or(OperandTypes.NUMERIC, OperandTypes.family(SqlTypeFamily.NUMERIC)),
```
Comment (xccui) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/sql/ScalarSqlFunctions.scala]: The `OperandTypes.NUMERIC` and `OperandTypes.family(SqlTypeFamily.NUMERIC)` are equivalent. Just use `OperandTypes.NUMERIC` here.

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/functions/ScalarFunctions.scala [hunk 2]
```diff
@@ -108,6 +108,17 @@ object ScalarFunctions {
     }
   }
 
+  /**
+    * Returns the logarithm of "x" with base "2".
```
Comment (xccui) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/functions/ScalarFunctions.scala]: ... with base **2**.

File: flink-libraries/flink-table/src/test/scala/org/apache/flink/table/expressions/ScalarFunctionsTest.scala [hunk 3]
```diff
@@ -553,6 +553,56 @@ class ScalarFunctionsTest extends ScalarTypesTestBase {
       math.log10(4.6).toString)
   }
 
+  @Test
+  def testLog2(): Unit = {
```
Comment (xccui) [inline flink-libraries/flink-table/src/test/scala/org/apache/flink/table/expressions/ScalarFunctionsTest.scala]: Could remove some redundant cases.

**Comment by yanghua** [comment]
@xccui thanks for your suggestion, have refactored, please review again.

**Comment by xccui** [comment]
Hi @yanghua, thanks for the update. I'll merge this. BTW, there's no need to squash the commits after a review 😄.

**Pull request closed without merging**